cmd/kubeadm/app/phases/addons/addons.go

```diff
@@ -45,8 +45,12 @@ func CreateEssentialAddons(cfg *kubeadmapi.MasterConfiguration, client clientset
 		return fmt.Errorf("error when parsing kube-proxy configmap template: %v", err)
 	}
 
+	repoPrefix := cfg.ImageRepository
+	if cfg.CIImageRepository != "" {
```

Comment:
I'd be open to add a `GetControlPlaneImageRepository()` helper func on `cfg` or something...